packages/flutter/lib/src/rendering/view.dart

```diff
@@ -67,10 +67,13 @@ class RenderView extends RenderObject with RenderObjectWithChildMixin<RenderBox>
   ///
   /// Typically created by the binding (e.g., [RendererBinding]).
   ///
-  /// The [configuration] must not be null.
+  /// Providing a [configuration] is optional, but a configuration must be set
```

Comment:
What is a RenderView without a provided configuration useful for?

Reply:
It allows us to decouple the creation of the RenderView (responsibility of the RawView element) from configuring it (responsibility of the RenderViewManager). Both must happen for the RenderView to ultimately be useful, but they are not required to happen at the same time and can be performed by different entities.

Reply:
Okay.  Maybe mention this in the docs.

Reply:
Although I haven't thoroughly understood the flow here, now that `RenderViewManager` is gone, is this separation still necessary?

Reply:
Oh I see, now `addRenderView` sets the configuration instead. Then why do we keep the parameter in the constructor? Is it just for backward compatibility?

Reply:
It was there originally and it's idiomatic style for RenderObjects that they take all properties as arguments. This makes RenderObjects a little more ergonomic to use if you want to use them directly without the widget layer.

And yes, it's also for backwards compatibility as there was no good reason to break it.